### PR TITLE
[Core] Improved bad error message.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

The current worker died message is really terrible. We need to provide more information.

Current; 
```
2021-05-06 17:27:36,254	WARNING worker.py:1115 -- A worker died or was killed while executing task a67dc375e60ddd1affffffffffffffffffffffff01000000.
```

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wuisawesome** [comment]
I definitely agree that we should improve this message, but I'm not sure this extra information will help. 

What if we did something along the lines of `A system error occurred while the task with id: <task id> was executing. See logs/worker-<worker-id>-<job_id>-<pid>.err` for more details. 

I think the node ip is already printed to the driver right?

**Comment by rkooo567** [comment]
I don't see node ip from this message; `2021-05-06 17:27:36,254	WARNING worker.py:1115 -- A worker died or was killed while executing task a67dc375e60ddd1affffffffffffffffffffffff01000000.
`

Hmm yeah if this is user-facing error messages, that type sounds better. But we might also need to tell users to ssh to nodes that have files.

**Comment by wuisawesome** [comment]
We don't attach the node ip when it executes on the local node, I guess we can just add that to the message to make that more clear for users I guess.

**Comment by rkooo567** [comment]
@wuisawesome I found something shocking... It seems like worker ID created from raylet is different from worker id of a worker log files; 

```
        assert job_id is None
        job_id = JobID.nil()
        # TODO(qwang): Rename this to `worker_id_str` or type to `WorkerID`
        worker.worker_id = _random_string()
```

That says, the log file we specify would be inaccurate.

File: src/ray/raylet/node_manager.cc [hunk 1]
```diff
@@ -1215,8 +1215,13 @@ void NodeManager::DisconnectClient(
...
         std::ostringstream error_message;
-        error_message << "A worker died or was killed while executing task " << task_id
-                      << ".";
+        error_message
+            << "A worker died or was killed while executing task by an unexepcted system "
+               "error. To troubleshoot the problem, check out a worker log. Task id: "
+            << task_id << " Worker id: " << worker->WorkerId()
+            << " Node id: " << self_node_id_
+            << " Worker IP address: " << worker->IpAddress()
+            << " Worker port: " << worker->Port();
```
Comment (clarkzinzow) [inline src/ray/raylet/node_manager.cc:1224]:
```suggestion
            << "A worker died or was killed while executing a task by an unexpected system "
               "error. To troubleshoot the problem, check the logs for the dead worker. Task ID: "
            << task_id << " Worker ID: " << worker->WorkerId()
            << " Node ID: " << self_node_id_
            << " Worker IP address: " << worker->IpAddress()
            << " Worker port: " << worker->Port();
```

**Review by clarkzinzow — APPROVED** [review]

**Comment by rkooo567** [comment]
@clarkzinzow @wuisawesome what's the best action here? 

1. Keep the current impl
2. Only specify the core worker logs
3. Fix the worker id in python different from core worker

We should eventually do 3, but I don't know if I want to do in this PR lol.

**Comment by wuisawesome** [comment]
> Only specify the core worker logs

Which ID do we currently have? If we can specify the correct log address, then I'd be in favor of that. Otherwise, lets' do (1), then (3), then (2)?

**Comment by rkooo567** [comment]
we only have ID for the core worker logs (not the python worker log which uses different ids). I think printing core worker logs are probably fine because if the worker crashes by system errors, it is mostly due to core worker, not the python

**Comment by rkooo567** [comment]
Actually it requires some work to be done in order to use the correct core worker logs because we need to write a correct language for files. I will just use the current format for now and probably do 3->2 in the future.

**Review by wuisawesome — APPROVED** [review]

**Comment by rkooo567** [comment]
There's no way the test_component_failure_2 is related to this PR. It also seems to be flaky in the master.

**Pull request merged**